### ISSUE TITLE
lintian: E: warewulf-nhc: dir-or-file-in-var-run var/run/nhc/

### ISSUE THREAD

**Issue by hmlth** [edf-hpc/warewulf-nhc#1]
E: warewulf-nhc: dir-or-file-in-var-run var/run/nhc/
N: 
N:    /var/run may be a temporary filesystem, so any directories or files
N:    needed there must be created dynamically at boot time.
N:  
N:    Refer to Debian Policy Manual section 9.3.2 (Writing the scripts) for
N:    details.
N:  
N:    Severity: serious, Certainty: possible
N:  
N:    Check: files, Type: binary, udeb
N:

**Comment by mehdid** [comment]
Only the first one is an issue, IMHO. It needs fixing the postinst to mkdir -p /var/run/nhc, AFAIK.

**Comment by hmlth** [comment]
I thinks what this error means is that it's useless to create /var/run/nhc in postinst because the whole /var/run tree can vanish on a a reboot and must be recreated automatically at runtime (like /tmp).

**Comment by rezib** [comment]
Yep, /var/run/\* subdirs must be handled with systemd-tmpfiles these days.

For example: https://github.com/edf-hpc/slurm-llnl/blob/debian/calibre9/debian/slurmctld.tmpfile

**Comment by roastercode** [comment]
@rezib your link is dead but https://www.freedesktop.org/software/systemd/man/systemd-tmpfiles.html would be a nice explanation with https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html# 

> Volatile and temporary files and directories are those located in /run (and its alias /var/run), /tmp, /var/tmp, the API file systems such as /sys or /proc, as well as some other directories below /var.
